### PR TITLE
add glob1 glob0 to glob

### PR DESCRIPTION
**Pull request by amstree**
- add glob1 glob0 to python2 stdlib glob

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Thanks for adding this! These functions aren't documented, but they're public and they've been preserved in Python 3, so I'm OK with adding them. Could you add them to the Python 3 stub too?

I don't think all the AnyStrs are right: these functions almost certainly work (in Python 2) if you pass them a mixture of str and unicode. Looking at https://hg.python.org/cpython/file/2.7/Lib/glob.py, I think the dirname arguments should just be `Union[str, unicode]` since they don't affect the return type. 

The Travis failure is due to an issue on our side that we'll hopefully fix soon.

File: stdlib/3/glob.pyi [hunk 1]
```diff
@@ -4,6 +4,12 @@
 from typing import List, Iterator, AnyStr
 import sys
 
+if sys.version_info >= (3, 6):
```
Comment (ambv) [inline stdlib/3/glob.pyi]: Why this check? `glob1` is still present on Python 3.6 and 3.7.

Comment (ambv) [inline stdlib/3/glob.pyi]: Oh, I see, it's because the name of the argument apparently changed.

File: stdlib/2/glob.pyi [hunk 2]
```diff
@@ -1,4 +1,8 @@
-from typing import List, Iterator, AnyStr
+from typing import List, Iterator, Union
 
-def glob(pathname: AnyStr) -> List[AnyStr]: ...
-def iglob(pathname: AnyStr) -> Iterator[AnyStr]: ...
+_string = Union[str, unicode]
+
+def glob(pathname: _string) -> List[_string]: ...
```
Comment (gvanrossum) [inline stdlib/2/glob.pyi]: I'm unhappy with this. Returning a list of unions is a rather big price to pay. And AFAICT glob() really does return a list of unicodes when the pattern is unicode, and a list of str when the pattern is str. @ambv maybe you can roll this back until we've sorted out what to do?

**Pull request merged**